File: RQ1_Prompt_Elements/Data/Scenario2/id_91.java
package Part2;


import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class IsBored {
    /**
     * * You'll be given a string of words, and your task is to count the number
     * of boredoms. A boredom is a sentence that starts with the word "I".
     * Sentences are delimited by '.', '?' or '!'.
   
     */
    public static int isBored(String s) {
        int count = 0;
        String[] sentences = s.split("[.!?]");
        for (String sentence : sentences) {
            String[] words = sentence.split(" ");
            boolean firstWord = false;
            for( String word : words){
                if(word.length()==0)
                    continue;
                else{
                    firstWord = true;
                    if (word.startsWith("I")) {
                        count++;
                    }
                }
                if(firstWord)
                    break;
            }
           
        }
        return count;
    }

}

File: HumanEvalPython/scenario2/id_91.py

def is_bored(S):
    """
    You'll be given a string of words, and your task is to count the number
    of boredoms. A boredom is a sentence that starts with the word "I".
    Sentences are delimited by '.', '?' or '!'.
   
    
    """

    import re
    sentences = re.split(r'[.?!]\s*', S)
    return sum(sentence[0:2] == 'I ' for sentence in sentences)
